packages/core/src/readers/LlamaParseReader.ts

```diff
@@ -130,14 +130,18 @@ export class LlamaParseReader extends FileReader {
   doNotCache?: boolean;
   // Wether to use a faster mode to extract text from documents. This mode will skip OCR of images, and table/heading reconstruction. Note: Non-compatible with gpt4oMode. Backend default is false.
   fastMode?: boolean;
-  // Wether to keep column in the text according to document layout. Reduce reconstruction accuracy, and LLM's/embedings performances in most cases.
+  // Wether to keep column in the text according to document layout. Reduce reconstruction accuracy, and LLM's/embedings performances in most cases. Backend default is false.
   doNotUnrollColumns?: boolean;
   // The page separator to use to split the text. Default is None, which means the parser will use the default separator '\\n---\\n'.
   pageSeperator?: string;
   // Whether to use gpt-4o to extract text from documents.
   gpt4oMode: boolean = false;
   // The API key for the GPT-4o API. Optional, lowers the cost of parsing. Can be set as an env variable: LLAMA_CLOUD_GPT4O_API_KEY.
   gpt4oApiKey?: string;
+  // Whether to split by page. Default separator: `\n---\n` Can be overwritten using splitByPageSeperator
+  splitByPage: boolean = true;
+  // The page separator used when splitByPage is true. Does not exist in Python version. Not actually sure what pageSeperator is for. pageSeperator gets sent to the LlamaParse API, while splitByPageSeperator is used in the getSubDocs function.
+  splitByPageSeperator: string = "\n---\n";
```

Comment:
we should just use whatever page seperator that LlamaParse is using - isn't that specified by the `pageSeperator` parameter? @hexapode can you help out here?

Reply:
It would be nice to get some clarification, `pageSeperator` is a param that gets send to the API Backend and is described as: 

> The page separator to use to split the text. Default is None, which means the parser will use the default separator '\\n---\\n'.

while `splitByPage` is an internal function described as:

> Whether to split by page (NOTE: using a predefined separator `\n---\n`)

 `pageSeperator` is not responsible for the seperator in `splitByPage`, so I have no clue what it actually does on the backend side. I'll draft this for now

Reply:
According to @hexapode, the JSON of the job result (https://docs.cloud.llamaindex.ai/llamaparse/API/get-job-json-result-api-parsing-job-job-id-result-json-get) already returns a collection of pages, so seems to me that we should handle that first

Reply:
That's already handled to the same extent as in python.
`loadJson` returns an array of json Objects containing `pages`, which represent a single file parsed. Each object under `pages` represent a single page and has a `page` field, next to `text`, `md`, etc.

`splitByPage` is only used in `loadData` method, so only for markdown or text and still not sure what about `pageSeperator`

Reply:
Related: https://github.com/run-llama/LlamaIndexTS/pull/1023

Reply:
Oh.. thank you, apparently, I don't know how to spell `separator` lol.